### PR TITLE
Add protocol messages for progress notifications

### PR DESCRIPTION
**Pull request by planger**
https://github.com/eclipse-glsp/glsp/issues/1019

### CODE REVIEW

File: content/documentation/protocol/content.md [hunk 1]
```diff
@@ -1217,11 +1217,19 @@ interface FitToScreenAction extends Action {
...
+Status notifications are typically shown directly on the diagram as an overlay for a certain time.
+Status messages, on the other hand, are persistent notifications and are typically shown in a dedicated notification area until the user discards them.
 
-This action is typically sent by the server to signal a state change. This action extends the corresponding Sprotty action to include a timeout. If a timeout is given the respective status should disappear after the timeout is reached.
+Progress notifications only appear while the long running operation is running and indicate the progress of this operation with messages and optionally a progress bar.
+
+##### 2.8.2.1. ServerStatusAction
+
+This action is typically sent by the server to signal a state change.
+This action extends the corresponding Sprotty action to include a timeout.
```
Comment (tortmayr) [inline content/documentation/protocol/content.md:1231]:
```suggestion
```

File: content/documentation/protocol/content.md [hunk 2]
```diff
@@ -1217,11 +1217,19 @@ interface FitToScreenAction extends Action {
...
+If a timeout is given the respective status should disappear after the timeout is reached.
```
Comment (tortmayr) [inline content/documentation/protocol/content.md:1232]:
```suggestion
```

**Review by tortmayr — CHANGES_REQUESTED** [review]

File: content/documentation/protocol/content.md (hunk 1 above)
Comment (planger) [reply content/documentation/protocol/content.md:1231]: The `ServerStatusAction` still has a `timeout` property. See https://github.com/eclipse-glsp/glsp-client/blob/master/packages/protocol/src/action-protocol/client-notification.ts#L40

Imho there was no reason to remove it, given that we are handling the status in glsp-client directly showing an overlay. Also it makes sense to have these two types of messages imho: 
- a single status message that can be shown at a given time to raise the awareness of something for a certain period
- messages, which are persistent and multiple ones can be collected over time

File: content/documentation/protocol/content.md (hunk 2 above)
Comment (planger) [reply content/documentation/protocol/content.md:1232]: See comment above.

**Review by tortmayr — APPROVED** [review]

**Pull request merged**